### ISSUE TITLE
Document WMS BGDI in API3 doc page?

### ISSUE THREAD

**Issue by loicgasser** [geoadmin/mf-chsdi3#2168]
Any reason not to document WMS BGDI geo services on api3 page?
ping @gjn , @davidoesch

**Comment by ltclm** [comment]
what besides a getcap link do you want to document?

**Comment by davidoesch** [comment]
Folks
The official documentation for all portrayal services are on the Cms : https://www.geo.admin.ch/en/geo-services/geo-services/portrayal-services-web-mapping.html

Cms (since migration is now done)
To do: 
-integrating listing  based on get cap for WMTS and WMS for all lang (task already assigned to Daniela...)
-Adding all FAQs
- rest services of api3: we should include it as iframe in the Cms 

Api3
Purely technical documentation with examples


Difference cms and api3 : the latter is for technicians only, goes beyond the detail required by the legal based mission of geo.admin.ch

**Comment by gjn** [comment]
I'm not a fan of including technical documentation as iframe in cms. We should use links. (and the css....). To cms visitor, api3 information is hard to find, this could be improved.

But, that doesn't mean we shouldn't/couldn't document wms in api3 as well. At least a list of available getcaps would make sense (as we do for wmts), describe the different projections, mention limitations, etc. A user starting with api3.geo.admin.ch won't ever know about the wms.

(Do we have access statistics of cms pages? I wonder how many people looks that catalog definitions in the cms...)

**Comment by davidoesch** [comment]
Cms is just what is required by the law - one single entry point. In 4 languages. Data content centric. And it is seo optimized. So we have to at least link api3 where  possible in the Cms. The reason we started api3: updating the Cms (old and new) is not  dev friendly.... And we would have to first do it in DE FR IT before we can't even do it in EN.
And yes - just last week a major external company doing SBB data integration was complaining that they miss the get cap content of WMTS in the Cms.

So I  would like to to revive the content in the new Cms .... And it is also on Daniela s to do list

**Comment by gjn** [comment]
New CMS is missing lots of information...

Anyway, I close this one. I think it could make sense to add WMS information on api3 page as well. @mariavkourouni or @Ariadnigaki could try to add it.

**Comment by ltclm** [comment]
how will we update these wms getcap caches?

**Comment by gjn** [comment]
What do you mean? In the CMS? or in the api3 doc? In api3 doc, it's onyl a linke to the wms getcap.

**Comment by ltclm** [comment]
if its only a link to the different wms getcap languages its already answered.
> And yes - just last week a major external company doing SBB data integration was complaining that they miss the get cap content of WMTS in the Cms.

but @davidoesch was speaking of the getcap content. I would just like to avoid that we have to manually update several getcap content caches in several systems after each wms deploy.

**Comment by gjn** [comment]
In old CMS, the getcap content was dynamially created using our official wmts getcap document (using javascript). But that's not possible anymore because of missing cms component....

But I agree with @ltclm , this should be automated in new CMS as well. Manual updates should be avoided mostly.

**Comment by ltclm** [comment]
yes, we should focus to document the things which are not visible in the getcap.
We should document that we have a getcap in several languages and how to acces them with url parameter ``&lang=(de|fr|it|rm|en)`` or as part of the uri path ``wms.geo.admin.ch/(de|fr|it|rm|en)/? `` [see here](https://github.com/geoadmin/wms-bgdi/pull/57).

**Comment by davidoesch** [comment]
in the new CMS Content of WMS is updated automatically [1] via GetCap...

[1] https://www.geo.admin.ch/de/geo-dienstleistungen/geodienste/darstellungsdienste-webmapping-webgis-anwendungen/web-map-services-wms.html

**Comment by loicgasser** [comment]
We could document and put a Getcap content in the api3 doc and put a link in the CMS until we can use javascript in the CMS or find a better solution.

**Comment by davidoesch** [comment]
JavaScript _can_ be used in our cms....

**Comment by gjn** [comment]
Yes, I was wrong.

I would still add the CMS in the api3 documentation. The same as for WMTS. WE could explain parameters, languages, etc as mentioned by @ltclm .

**Comment by procrastinatio** [comment]
What is the status of this ?

**Comment by davidoesch** [comment]
Again:

Central entry point is the CMS geo.admin.ch, in four lang. the reason we had api3 is to have pure technical documentation in EN only.
other arguemnts
https://github.com/geoadmin/mf-chsdi3/issues/2168#issuecomment-232319996
https://github.com/geoadmin/mf-chsdi3/issues/2168#issuecomment-232477850

Status
-> it is task listed in CMS to do list. but depends of prio of worload for @pauloamado

**Comment by ltclm** [comment]
still pending?

**Comment by pauloamado** [comment]
I will talk with @davidoesch to get more infos

**Comment by davidoesch** [comment]
On hold. We will restructure api doc with re4
